src/pyk/proof/reachability.py

```diff
@@ -163,18 +179,38 @@ def summary(self) -> Iterable[str]:
 
 class APRProver:
     proof: APRProof
+    kcfg_explore: KCFGExplore
     _is_terminal: Callable[[CTerm], bool] | None
     _extract_branches: Callable[[CTerm], Iterable[KInner]] | None
 
+    main_module_name: str
+    some_dependencies_module_name: str
+    all_dependencies_module_name: str
+
     def __init__(
         self,
         proof: APRProof,
+        kcfg_explore: KCFGExplore,
         is_terminal: Callable[[CTerm], bool] | None = None,
         extract_branches: Callable[[CTerm], Iterable[KInner]] | None = None,
     ) -> None:
         self.proof = proof
+        self.kcfg_explore = kcfg_explore
         self._is_terminal = is_terminal
         self._extract_branches = extract_branches
+        self.main_module_name = self.kcfg_explore.kprint.definition.main_module_name
+
+        self.some_dependencies_module_name = self.main_module_name + '-DEPENDS-MODULE'
+        self.kcfg_explore.add_dependencies_module(
+            self.main_module_name,
+            self.some_dependencies_module_name,
+            [c for c in proof.dependencies if c.label != self.proof.id],
+            priority=1,
+        )
+        self.all_dependencies_module_name = self.main_module_name + '-CIRCULARITIES-MODULE'
+        self.kcfg_explore.add_dependencies_module(
+            self.main_module_name, self.all_dependencies_module_name, proof.dependencies, priority=1
+        )
```

Comment:
```suggestion
        self.some_dependencies_module_name = self.main_module_name + '-DEPENDS-MODULE'
        self.kcfg_explore.add_dependencies_module(
            self.main_module_name,
            self.some_dependencies_module_name,
            [build_claim(c) for c in proof.dependencies if c.label != self.proof.id],
            priority=1,
        )
        self.all_dependencies_module_name = self.main_module_name + '-CIRCULARITIES-MODULE'
        self.kcfg_explore.add_dependencies_module(
            self.main_module_name, self.all_dependencies_module_name, proof.dependencies, priority=1
        )
```

Now since we're storing `APRProof`, we need to build the claim right at the moment we are sending it as a module to the backend.